jib-plugins-common/src/main/java/com/google/cloud/tools/jib/plugins/common/DefaultCredentialRetrievers.java

```diff
@@ -151,8 +152,11 @@ public DefaultCredentialRetrievers setCredentialHelper(@Nullable String credenti
       // If credential helper contains file separator, treat as path; otherwise treat as suffix
       if (credentialHelper.contains(FileSystems.getDefault().getSeparator())) {
         if (!Files.exists(Paths.get(credentialHelper))) {
-          throw new FileNotFoundException(
-              "Specified credential helper was not found: " + credentialHelper);
+          String osName = systemProperties.getProperty("os.name").toLowerCase(Locale.ENGLISH);
+          if (!osName.contains("windows") || !Files.exists(Paths.get(credentialHelper + ".cmd"))) {
+            throw new FileNotFoundException(
+                "Specified credential helper was not found: " + credentialHelper);
+          }
```

Comment:
Would it make sense to push this into `DockerCredentialHelper` or `credentialRetrieverFactory.dockerCredentialHelper()`?

Reply:
The logic here is interpreting ahead the given configuration value to determine if it is a full path or a credential helper suffix. This semantics is tightly coupled with how we defined the `<credHelper>` configuration, so it cannot make into jib-core.

Moreover, `DockerCredentialHelper` by design doesn't do any pre-validation (including file checking); it is OK even if the credential helper binary doesn't exist or work when another credential retriever with a higher precedence provides credentials in the end.